File: src/main/java/net/dontdrinkandroot/gitki/wicket/component/bspanel/index/IndexFilePanel.java
package net.dontdrinkandroot.gitki.wicket.component.bspanel.index;

import net.dontdrinkandroot.gitki.model.FilePath;
import net.dontdrinkandroot.gitki.wicket.model.AbstractPathNameModel;
import net.dontdrinkandroot.wicket.bootstrap.component.panel.PanelHeading;
import net.dontdrinkandroot.wicket.bootstrap.component.panel.PlainPanel;
import net.dontdrinkandroot.wicket.component.basic.Heading;
import org.apache.wicket.Component;
import org.apache.wicket.model.IModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public abstract class IndexFilePanel extends PlainPanel<FilePath>
{
    public IndexFilePanel(String id, IModel<FilePath> model)
    {
        super(id, model);
    }

    @Override
    protected Component createHeading(String id)
    {
        return new PanelHeading(id, new AbstractPathNameModel(this.getModel()), Heading.Level.H2);
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/component/bspanel/index/IndexFilePanel.kt
package net.dontdrinkandroot.gitki.wicket.component.bspanel.index

import net.dontdrinkandroot.gitki.model.FilePath
import net.dontdrinkandroot.gitki.wicket.model.AbstractPathNameModel
import net.dontdrinkandroot.wicket.bootstrap.component.card.PlainCard
import net.dontdrinkandroot.wicket.component.basic.Heading
import org.apache.wicket.model.IModel

abstract class IndexFilePanel(id: String, model: IModel<FilePath>) : PlainCard<FilePath>(id, model) {

    override fun createHeader(id: String) = Heading(id, AbstractPathNameModel(this.model), Heading.Level.H2)
}